background/providers/rss/tests/parse-feeds.test.js

```diff
@@ -0,0 +1,79 @@
+/* eslint guard-for-in: 0 max-nested-callbacks: 0 */
+
+import {test} from 'ava';
+
+import {default as ParseFeed} from '../lib/parse-feed';
+
+import {
+    generateAtomFeed,
+    generateTorrentNamespacedFeed,
+    generateTorrentFeed,
+    generateMixedTorrentFeed
+} from './generate-test-data';
+import {default as torrents} from './testdata';
+
+const feeds = [generateAtomFeed, generateTorrentNamespacedFeed, generateTorrentFeed, generateMixedTorrentFeed];
+
+feeds.forEach(fn => {
+    fn().then(feed => {
+        test.cb(`The ${fn.name} feed should have ${torrents.length} elements`, t => {
+            let _count = 0;
+            new ParseFeed(feed).on('torrent', () => {
+                _count++;
+            }).on('end', () => {
+                t.is(_count, torrents.length);
+                t.end();
+            });
+        });
+    });
+});
+
+test.cb('The generateMixedTorrentFeed feed should contain "item" objects', t => {
```

Comment:
It's done this way because the ParseFeed doesn't return any Promises it instead emits events, so instead of aggregating all of the torrents in a feed and returning them we emit a 'torrent' event every time we've parsed a torrent.

The test could still probably be re-written to look clearer though (and not break xo without disabling 'guard-for-in' and 'max-nested-callbacks'). Any suggestions?